spec/features/school_class/listing_school_classes_spec.rb

```diff
@@ -3,110 +3,235 @@
 require 'rails_helper'
 
 RSpec.describe 'Listing school classes', type: :request do
-  before do
-    authenticated_in_hydra_as(owner)
-    stub_user_info_api_for_users([teacher, owner_teacher].map(&:id), users: [owner_teacher, teacher])
-
-    create(:class_student, school_class:, student_id: student.id)
-  end
-
   let(:headers) { { Authorization: UserProfileMock::TOKEN } }
-  let!(:school_class) { create(:school_class, name: 'Test School Class', teacher_ids: [teacher.id], school:) }
   let(:school) { create(:school) }
   let(:student) { create(:student, school:) }
   let(:teacher) { create(:teacher, school:, name: 'School Teacher') }
   let(:owner) { create(:owner, school:) }
-
   let(:owner_teacher) { create(:teacher, school:, id: owner.id, name: owner.name, email: owner.email) }
-  let!(:owner_school_class) { create(:school_class, name: 'Owner School Class', teacher_ids: [owner_teacher.id], school:) }
 
-  it 'responds 200 OK' do
-    get("/api/schools/#{school.id}/classes", headers:)
-    expect(response).to have_http_status(:ok)
-  end
-
-  it 'responds with the school classes JSON' do
-    get("/api/schools/#{school.id}/classes", headers:)
-    data = JSON.parse(response.body, symbolize_names: true)
+  let!(:school_class) { create(:school_class, name: 'Test School Class', teacher_ids: [teacher.id], school:) }
+  let!(:owner_school_class) { create(:school_class, name: 'Owner School Class', teacher_ids: [owner_teacher.id], school:) }
 
-    expect(data.first[:name]).to eq('Test School Class')
+  before do
+    authenticated_in_hydra_as(owner)
+    stub_user_info_api_for_users([teacher, owner_teacher].map(&:id), users: [owner_teacher, teacher])
+    create(:class_student, school_class:, student_id: student.id)
   end
 
-  it 'only responds with the user\'s classes if the my_classes param is present' do
-    get("/api/schools/#{school.id}/classes?my_classes=true", headers:)
-    data = JSON.parse(response.body, symbolize_names: true)
-
-    expect(data.first[:name]).to eq(owner_school_class.name)
+  # Helper to make API call and parse response
+  def get_classes(path = "/api/schools/#{school.id}/classes")
+    get(path, headers:)
+    JSON.parse(response.body, symbolize_names: true)
   end
 
-  it 'responds with the teachers JSON' do
-    get("/api/schools/#{school.id}/classes", headers:)
-    data = JSON.parse(response.body, symbolize_names: true)
-    expect(data.first[:teachers].first[:name]).to eq('School Teacher')
+  def find_class(data, name)
+    data.find { |c| c[:name] == name }
   end
```

Comment:
could name these variables a little more semantically to clarify how this method will be used